### ISSUE TITLE
Obtain absolute transmission from the caxis plot

### ISSUE THREAD

**Issue by NolannR** [kklmn/xrt#111]
Hi,

I know how to get total transmission on a single bin using plots[0].caxis.total1D method, therefore I do have total transmission for each bins on the plot,  and I also do know how to calculate the absolute transmission on a monoenergetic source. But I would like to know how to obtain absolute transmission for each beam using the flux and the Nray alive.

Thank you in advance for your help.

**Comment by kklmn** [comment]
Hi,

What do you mean by "absolute transmission"? I would think of it as *flux_after_interaction/flux_before_interaction* for whatever interaction of your interest, and you do have the flux values as you say.

**Comment by NolannR** [comment]
During a run, I obtain a certain e_histogram. For a simple periodic coating I used, I obtain 14.2%. For a monoenergetic run, using the flux and nrays I can calculate it. But for a polyenergetic run, I do have my histogram with absolute value, and the total flux is calculated with all the bins. Simply divide the flux for for a single energy by the Nray alive in total gives me a much lower transmission.

**Comment by kklmn** [comment]
What's your physical definition?

**Comment by NolannR** [comment]
![Image_in_the_detector_plane](https://user-images.githubusercontent.com/67743260/167091207-d8893eb7-141b-4932-b68d-aab403d964e9.png)

Perhaps with an image, it will be easier to explain.

For a polychromatic source, I have this histograms for energy. If we sum each value for each beam, we obtain the flux. For this example, I have around 170 (at maximum) for 1140 flux. This gives me around 15-15.2%.
But for a monochromatic source, at this energy, (single non-zero bins), I have 14.2%. The difference looks important to me. Otherwise maybe it is just because there are not enough rays. Correct me if I'm wrong.

**Comment by kklmn** [comment]
ok, I see you can't define what you want to get in physical terms. And this is exactly your problem here. When you know what you want to get, you will find a way to calculate it. Now you are just doing some exercises with the obtained arrays.
The quantity you calculate (bin height over the histogram sum) depends on the bin width. If you, say, halve it, you will get roughly half a number of your percentage (don't know what it is), so what?

**Comment by NolannR** [comment]
What I want is the absolute transmission of my instrument. I can obtain it for each monoenergetic ray with a generator, but what I would like to obtain is the same thing but from the energy histogram. The % above is the transmission of my instrument at a certain energy using two periodic multilayers coatings. For a monoenergetic run around 16 keV, it gives me around 14%. I would like to obtain the same percentage from the e histogram on a polychromatic source.

**Comment by kklmn** [comment]
> The % above is the transmission of my instrument at a certain energy

Wrong. ..._into a certain bandwidth_ (your bin width) centered at a certain energy. A comparison of a zero-bandwidth result with a finite bandwidth result may have a meaning that you should know as *you* are doing it. For me, this meaning is not obvious.

**Comment by NolannR** [comment]
Ok, I want to run two scripts to get the transmission at my peak energy. To do so, I want to use afterscript method in run_ray_tracing, because I want to change the repeats (I want to lower it, I need a huge number of ray_tracing for uniformity, but a few is enough to get transmission). In generator I can run both but with the same number of repeats.

Using afterscript and calling the function, I am able to open a second window after the first run. However, nothing appears.

Here's the script : 

------------------------------------------------------------------------------------------------------------------------------
def tracing2() : 
    xrtrun.run_ray_tracing(plots=plots, repeats=3, generator=gen2, backend=r"raycing", beamLine=beamline())
   
def main():
    xrtrun.run_ray_tracing(plots=plots(), repeats=4, generator=gen1, beamLine=beamline(), afterScript=tracing2())
    
if __name__ == '__main__':
    main()
------------------------------------------------------------------------------------------------------------------------------

Basically, the thing I want to do is to have a repeats that is different from the previous one to decrease the time of calculus.

Thank you in advance

**Comment by kklmn** [comment]
Search the examples `06_AnalyzerBent1D` and `07_AnalyzerBent2D` for how repeats can be changed.

**Comment by NolannR** [comment]
Ok, by using set_repeats method in generator just before yield.

I obtained what I wanted.

Thank you a lot !
